guides/expression_language.md

```diff
@@ -166,6 +115,158 @@ let x = if decision { 42 }; // no else branch defaults to '()'
+
+Like C, arrays are accessed with zero-based, non-negative integer indices:
+
+> _array_ `[` _index position from 0 to length−1_ `]`
+
+```rust
+let some_list = ["foo", "bar", 42];
+
+let second_element = some_list[1];
+
+// second_element is "bar"
+```
+
+### Access Element From end
+
+A _negative_ position accesses an element in the array counting from the _end_, with −1 being the
+_last_ element.
+
+> _array_ `[` _index position from −1 to −length_ `]`
+
+```rust
+let some_list = ["foo", "bar", 42];
+
+let second_element = some_list[-2];
+let last_element = some_list[-1];
+
+// second_element is "bar"
+// last_element is 42
+```
+
+| Function | Parameter(s)                       | Description |
+| -------- | ------------                       | ----------- |
+| `get`    | position, counting from end if < 0 | gets a copy of the element at a certain position (`()` if the position is not valid) |
+| `len`    | _none_                             | returns the number of elements |
+| `filter` | predicate (usually a closure) | constructs a new array with all items that return `true` when called with the predicate function taking the following parameters:<ol><li>array item</li><li>_(optional)_ offset position</li></ol> |
+| `all`    | predicate (usually a closure) | returns `true` if all items return `true` when called with the predicate function taking the following parameters:<ol><li>array item</li><li>_(optional)_ offset position</li></ol>  |
+
+Examples
+```rust
+let some_list = [1, 2, 3, 4, "foo", "bar"];
+
+let foo = some_list.get(4); // "foo"
+
+let items_count = some_list.len(); // 6
+
+let only_foo_and_bar = some_list.filter(|item| item == "foo" || item == "bar"); // ["foo", "bar"]
+// let only_foo_and_bar = some_list.filter(|item, idex_in_array| item == "foo" || item == "bar");
+
+let another_list = [3, 5, 7, 9, 10, 20, 30];
+
+let all_greater_than_2 = another_list.all(|item| item > 2); // true
+let all_greater_than_10 = another_list.all(|item| item > 10); // false
+// let all_greater_than_10 = another_list.all(|item, idex_in_array| item > 10); 
+```
+
+
+## Maps
+
+Maps are hash dictionaries. Properties are all dynamic values and can be freely added and retrieved.
+
+Map literals are built within braces `#{` ... `}` with _name_`:`_value_ pairs separated by
+commas `,`:
+
+> `#{` _property_ `:` _value_`,` ... `,` _property_ `:` _value_ `}`
+>
+> `#{` _property_ `:` _value_`,` ... `,` _property_ `:` _value_ `,` `}`     `// trailing comma is OK`
+
+```rust
+let some_map = #{              // map literal with 2 properties
+    foo: 42,
+    bar: "hello",
+};
+```
+### Dot notation
+
+The _dot notation_ allows to access properties by name.
+
+> _object_ `.` _property_
+
+```rust
+let some_map = #{              // map literal with 2 properties
+    foo: 42,
+    bar: "hello",
+};
+
+some_map.foo // 42
+some_map.bar // "hello"
+
+```
+
+### Non-existing property
+
+Trying to read a non-existing property returns `()` instead of causing an error.
```

Comment:
this actually raises in our expression language

Reply:
Ah thanks :heart: ! Changed!